src/FsCodec/StreamName.fs

```diff
@@ -128,3 +128,20 @@ module StreamName =
     /// Throws <c>InvalidArgumentException</c> if the stream name is not well-formed</summary>
     /// <remarks>Inverse of <c>create</c></remarks>
     let (|CategoryAndIds|) : StreamName -> struct (string * string[]) = splitCategoryAndIds
+
+    module Parse =
+        let parse category f (name: StreamName) =
```

Comment:
blank line, inside module and at EOF
can you do a 4th one too - streamId ones have 4
also why no pattern match to pull the ids out of the array and validate len ? (can use `CategoryAndIds` pattern)